### ISSUE TITLE
Can only use custom font on one canvas

### ISSUE THREAD

**Issue by jrc03c** [processing/p5.js#1097]
I'm trying to use a custom font on two different canvases, but when I do, the `canvas2.text()` function only seems to draw stuff to the original canvas. However, if I disable the `canvas2.textFont()` call, the text appears correctly on the second canvas, but *not* rendered in its custom font! (NOTE: I've created the second canvas using `createGraphics()` instead of `createCanvas()`.) Here's the code:

```
var myFont;

function preload(){
    myFont = loadFont("https://dl.dropboxusercontent.com/u/69679202/roboto.ttf");
}

function setup(){
    var canvas1 = createCanvas(200, 200);
    background(0);
    fill(255);
    textFont(myFont);
    textSize(32);
    text("Hello, world!", 10, 24);
    
    var canvas2 = createGraphics(200, 200);
    canvas2.style("display", "block");
    canvas2.background(255, 0, 0);
    canvas2.fill(0);
    canvas2.textFont(myFont);
    canvas2.textSize(32);
    canvas2.text("Goodbye, world!", 10, 24);
}
```

And here's the result:

![screen shot 2015-11-11 at 7 55 05 am](https://cloud.githubusercontent.com/assets/255934/11092503/9c120fa2-8849-11e5-8f57-c0f79fa18963.png)

Am I doing something wrong, or is this a bug? Thanks!

**Comment by jrc03c** [comment]
Actually, I'm beginning to wonder if this is a canvas problem, not a p5js problem. I tried setting the font the old-fashioned way (see below), but I got the same result...

```
function setup(){
    noCanvas();
    
    var canvas1 = createGraphics(200, 200);
    canvas1.id("canvas1");
    canvas1.style("display", "block");
    canvas1.background(0);
    canvas1.fill(255);
    document.getElementById("canvas1").getContext("2d").font = "20px Times New Roman";
    canvas1.text("Hello, world!", 0, 20);
    
    var canvas2 = createGraphics(200, 200);
    canvas2.id("canvas2");
    canvas2.style("display", "block");
    canvas2.background(255, 0, 0);
    canvas2.fill(0);
    document.getElementById("canvas2").getContext("2d").font = "20px Comic Sans";
    canvas1.text("Goodbye, world!", 0, 20);
}
```

![screen shot 2015-11-11 at 10 43 13 am](https://cloud.githubusercontent.com/assets/255934/11096700/0f516b18-8861-11e5-8fe6-09729f829b31.png)

**Comment by jrc03c** [comment]
In the meantime, I've been able to set the fonts "manually" by accessing each canvas context's `font` property, like this:

```
function setup(){
    noCanvas();
    
    var canvas1 = createGraphics(200, 200);
    canvas1.id("canvas1");
    canvas1.style("display", "block");
    canvas1.background(0);
    canvas1.fill(255);
    document.getElementById("canvas1").getContext("2d").font = "20px Arial";
    canvas1.text("Hello, world!", 0, 20);
    
    var canvas2 = createGraphics(200, 200);
    canvas2.id("canvas2");
    canvas2.style("display", "block");
    canvas2.background(255, 0, 0);
    canvas2.fill(0);
    document.getElementById("canvas2").getContext("2d").font = "20px Georgia";
    canvas2.text("Goodbye, world!", 0, 20);
}
```

![screen shot 2015-11-11 at 10 54 15 am](https://cloud.githubusercontent.com/assets/255934/11097013/95be98c8-8862-11e5-8127-c3b96bf27398.png)

**Comment by mikewesthad** [comment]
I took a look at this since it seemed related to another issue - #1135 - and I do think they are tangentially related.

From what I saw, when an open type font is loaded (like roboto.ttf), the font is rendered by p5.Font's [_renderPath(...)](https://github.com/processing/p5.js/blob/master/src%2Ftypography%2Fp5.Font.js#L286) method.  This reads out the outline information from the font and draws it using lines and curves.  (In contrast, system fonts are drawn by letting the canvas handle things using [ctx.fillText(...)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillText).)  The problem here is that the _renderPath method finds the canvas context that it needs by asking the font for its associated p5 renderer & context ([here](https://github.com/processing/p5.js/blob/master/src%2Ftypography%2Fp5.Font.js#L290)).  This would be your first canvas, the one created using createCanvas(...).  So it seems that no matter what, if you use `textFont(myFont)` to draw an external font, that will set things up to draw to the first canvas.

That seems like a bug.  Maybe the _renderPath(...) method should be passed the renderer and drawing context as arguments?  Or things could be rearranged so that the textFont(...) method configures the Font object with the appropriate renderer and context?

**Workaround suggestion**

If you use a font already on your system, like `canvas2.textFont("Arial")`, everything should work fine  for drawing to canvas2 because this doesn't use _renderPath(...).  But that workaround doesn't help you when loading an external ttf and using two canvases.  You could do this instead:

Bring your font into your project using an [font-face](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face) in your CSS, like this:

```css
@font-face {
    font-family: 'Roboto Condensed';
    src: url('https://dl.dropboxusercontent.com/u/69679202/roboto.ttf');
}
```

Then your sketch will work (with a minor tweak):

```js

var canvas2;

function setup(){
    createCanvas(200, 200);
    fill(255);
    textSize(32);    

    canvas2 = createGraphics(200, 200);
    canvas2.elt.style.display = "block";
    canvas2.fill(0);
    canvas2.textSize(32);
}

function draw() {
    // Font-faces take a little while to load.  This will redraw
    // every frame trying to use Roboto Condensed.  Until Roboto
    // has been loaded, a fallback font will be used.  There are 
    // probably more elegant ways to do this, depending on your project. 

    background(0);
    textFont("Roboto Condensed");
    text("Hello, world!", 20, 40);

    canvas2.background(255, 0, 0);
    canvas2.textFont("Roboto Condensed");
    canvas2.text("Hello, world!", 20, 40);
}
```

![screenshot 2015-12-05 16 49 19](https://cloud.githubusercontent.com/assets/1131802/11610665/313fe82c-9b70-11e5-8804-e828c03b21b0.png)

**Comment by jrc03c** [comment]
@mikewesthad That workaround seems acceptable to me. Thanks for your help!

**Comment by dhowe** [comment]
This should be fixed with pr #1168 

![screen shot 2015-12-09 at 6 34 26 pm](https://cloud.githubusercontent.com/assets/737638/11683170/338c9652-9ea4-11e5-9ef5-9a355539742a.png)

**Comment by lmccart** [comment]
fixed with #1168

**Comment by dhowe** [comment]
See also #1413 for a better fix
